src/pipx/venv.py

```diff
@@ -173,21 +176,20 @@ def install_package(
         is_main_package: bool,
     ) -> None:
 
-        with animate(f"installing package {package_or_url!r}", self.do_animation):
-            if pip_args is None:
-                pip_args = []
+        if pip_args is None:
+            pip_args = []
+        if package is None:
+            # If no package name is supplied, install only main package
+            #   first in order to see what its name is
+            package = self.install_package_no_deps(package_or_url, pip_args)
+            # TODO 20191228: should we just fail if we cannot determine package?
```

Comment:
I have a TODO here we should resolve.  I'm thinking if we cannot determine package name here we should just fail at this point before we try to install anything else.  Does that sound good?

Reply:
Not sure if I need to start a review or not (github mechanics) but mostly I'm asking @cs01 and any other interested parties.